Ast/src/Parser.cpp

```diff
@@ -1622,11 +1626,17 @@ AstTypeOrPack Parser::parseTypeOrPack()
 
 AstType* Parser::parseType(bool inDeclarationContext)
 {
+    Location begin = lexer.current().location;
+
+    Lexeme::Type c = lexer.current().type;
+    if (c == '|' || c == '&')
+    {
+        return parseTypeSuffix(nullptr, begin);
```

Comment:
This will increment the recursion count without resetting it back to its old value.

Reply:
fixed!